### ISSUE TITLE
NextAuth example works locally but fails on Zeit

### ISSUE THREAD

**Issue by shea256** [iaincollins/next-auth#96]
Hey all,

In the NextAuth example (https://github.com/iaincollins/next-auth/tree/master/example), the file `pages/auth/index.js` has the following code:

```js
import React from 'react'
import Router from 'next/router'
import { NextAuth } from 'next-auth/client'
import Layout from '../../components/layout'

export default class extends React.Component {
  
  static async getInitialProps({req}) {
    return {
      session: await NextAuth.init({req}),
      linkedAccounts: await NextAuth.linked({req}),
      providers: await NextAuth.providers({req})
    }
  }

  render() {
    ...
  }
}
```

This page works perfectly fine when running locally, but unfortunately fails when deployed to Zeit using now.

In the deployment, I get the following error:

<img width="1161" alt="Screen Shot 2020-03-31 at 12 31 54 PM" src="https://user-images.githubusercontent.com/994646/78070747-53ed2700-736a-11ea-9efb-9819866578c4.png">

I tried troubleshooting this to no avail.

I saw this note in the NextAuth documentation:

> CSRF, Session and Linked Account endpoints are provided for Single Page Apps.

Based on this, I had a suspicion that `NextAuth.linked` and `NextAuth.providers` shouldn't be called on the server and should only be called on the client, so I modified the code to only call these functions on the client.

Unfortunately, this didn't fix the problem. The page still shows a server error when it is loaded on the client.

Would appreciate any help, and would be great to find out if this is an issue with NextAuth, Next.js, Zeit, or one of my minor modifications.

Thanks!

**Comment by iaincollins** [comment]
Unfortunately, NextAuth hasn't been updated to work well with more recent versions of Next.js.

It's still possible, but it's tricky to get working.

The good news, is I've just made [an announcement about version 2.0](#99) which is designed for recent versions of Next.js and Now.sh and will be much easier to get up and running.

This probably isn't that helpful right now, but if you can hold off implementing for 3-4 weeks you should find the new version much easier to use.

**Comment by iaincollins** [comment]
Closing this off as addressed by version 2.0.
